lib/device_info/CMakeLists.txt

```diff
@@ -0,0 +1,10 @@
+#
+# Copyright (c) 2019 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+if(CONFIG_DEVICE_INFO)
+    zephyr_library()
+    zephyr_library_sources(device_info.c)
```

Comment:
Please use the more conventional `zephyr_library_sources_ifdef` and get rid of the `if` condition.